.clang-tidy

```diff
@@ -12,13 +12,21 @@ Checks: |
   -cppcoreguidelines-pro-bounds-pointer-arithmetic
   -cppcoreguidelines-init-variables
   -cppcoreguidelines-pro-type-member-init
+  -cppcoreguidelines-avoid-do-while
+  -cppcoreguidelines-pro-type-const-cast
+  -cppcoreguidelines-pro-bounds-array-to-pointer-decay
+  -cppcoreguidelines-avoid-const-or-ref-data-members
+  -cppcoreguidelines-pro-bounds-constant-array-index
   bugprone-*
   -bugprone-sizeof-expression
   -bugprone-narrowing-conversions
   -bugprone-macro-parentheses
+  -bugprone-easily-swappable-parameters
+  -bugprone-implicit-widening-of-multiplication-result
```

Comment:
✅ Most of these seem to be about comparing a `std::size_t` against a `celeritas::size_type`:
```
/home/runner/work/celeritas/celeritas/src/celeritas/field/RZMapFieldParams.cc:47:31: warning: performing an implicit widening conversion to type 'unsigned long' of a multiplication performed in type 'unsigned int' [bugprone-implicit-widening-of-multiplication-result]
        inp.field_z.size() == inp.num_grid_z * inp.num_grid_r,
```